pack/entrypoint.sh

```diff
@@ -61,18 +58,33 @@ dotnet new classlib --name $PACKAGE_NAME --output $SRC/$PACKAGE_NAME --framework
 rm -f ./$SRC/$PACKAGE_NAME/Class1.cs
 
 echo "Adding Protobuf files..."
-found=$(find $PROTOBUF_FOLDER -name "*.proto" | grep -q '.' && echo 'found')
-
-if [ "$found" = "found" ]; then
-  for file in $(find $PROTOBUF_FOLDER -name "*.proto")
-  do
-      path="../.$file"
-      dotnet-grpc add-file  --services None --project ./$SRC/$PACKAGE_NAME/$PACKAGE_NAME.csproj  $path
-  done
 
-  echo "Packing $PACKAGE_NAME version $PACKAGE_VERSION at $OUTPUT_PATH"...
-  dotnet pack ./$SRC/$PACKAGE_NAME/$PACKAGE_NAME.csproj -c Release -p:PackageVersion=$PACKAGE_VERSION -o $OUTPUT_PATH
-else
+count=$(find $PROTOBUF_FOLDER -name "*.proto" | wc -l)
+if [ "$count" -eq "0" ]; then
   echo "Error: No Protobuf file found at '$PROTOBUF_FOLDER'"
   exit -1
+fi  
+
+for file in $(find $PROTOBUF_FOLDER -name "*.proto" -exec readlink -f {} \;)
+do
+  echo "Protobuf file found at '$file'. Trying to add it to the project..."
+  dotnet-grpc add-file  --services None --project $PROJ  $file
+done
+
+added_files=$(dotnet-grpc list --project $PROJ | grep -c "$PROTOBUF_FOLDER" )
+if [ $added_files -ne $count ]; then
+  echo "Error: The number of the Protobuf files found at the given path and the number of added files to the project doesn't march (found=$count,added=$added_files)!"
```

Comment:
Why would that ever happen?

Reply:
Just an approval process to ensure nothing left behind